lib/Sema/CSDiagnostics.cpp

```diff
@@ -5558,11 +5581,14 @@ bool ArgumentMismatchFailure::diagnoseUseOfReferenceEqualityOperator() const {
   // let's avoid producing a diagnostic second time, because first
   // one would cover both arguments.
   if (getAsExpr(getAnchor()) == rhs && rhsType->is<FunctionType>()) {
-    auto &cs = getConstraintSystem();
-    if (cs.hasFixFor(getConstraintLocator(
-            binaryOp, {ConstraintLocator::ApplyArgument,
-                       LocatorPathElt::ApplyArgToParam(
-                           0, 0, getParameterFlagsAtIndex(0))})))
+    auto *argLoc = getConstraintLocator(
+        binaryOp,
+        {ConstraintLocator::ApplyArgument,
+         LocatorPathElt::ApplyArgToParam(0, 0, getParameterFlagsAtIndex(0))});
+
+    if (llvm::any_of(getSolution().Fixes, [&argLoc](const ConstraintFix *fix) {
```

Comment:
I wonder if this is a common enough operation to worth a `Solution::hasFixForLocator()` like I think in `ConstraintSystem`  ... maybe :)

Reply:
I think that's something we shouldn't encourage in the diagnostic phase because all of the diagnostics should be isolated, but there are a couple of exceptions at the moment just because we don't have a good answer for them in the solver...

Reply:
Makes sense :)